ambry-store/src/test/java/com.github.ambry.store/BlobStoreStatsTest.java

```diff
@@ -241,8 +315,351 @@ public void testValidDataSizeAfterDeletes() throws InterruptedException, StoreEx
   }
 
   /**
-   * Test the static method that converts the quota stats in the form of a nested {@link Map} to the generic
-   * {@link StatsSnapshot} object.
+   * Basic test to verify that the {@link BlobStoreStats} can scan the index, populate the buckets and use these buckets
+   * to report stats correctly.
+   * @throws StoreException
+   * @throws InterruptedException
+   * @throws IOException
+   */
+  @Test
+  public void testBucketingBasic() throws StoreException, InterruptedException, IOException {
+    assumeTrue(bucketingEnabled);
+    final CountDownLatch scanStartedLatch = new CountDownLatch(1);
+    MockThrottler mockThrottler = new MockThrottler(scanStartedLatch, new CountDownLatch(0));
+    throttlers.put(BlobStoreStats.IO_SCHEDULER_JOB_TYPE, mockThrottler);
+    // add 3 puts with expiry
+    long expiresAtInMs = state.time.milliseconds() + TimeUnit.SECONDS.toMillis(30);
+    state.addPutEntries(3, CuratedLogIndexState.PUT_RECORD_SIZE, expiresAtInMs);
+    long logSegmentForecastOffsetMs = state.time.milliseconds();
+    int bucketCount = 2 * (int) (logSegmentForecastOffsetMs / BUCKET_SPAN_IN_MS);
+    BlobStoreStats blobStoreStats = setupBlobStoreStats(bucketCount, logSegmentForecastOffsetMs);
+    // proceed only when the scan is started
+    assertTrue("IndexScanner took too long to start", scanStartedLatch.await(5, TimeUnit.SECONDS));
+    verifyAndGetContainerValidSize(blobStoreStats, state.time.milliseconds());
+    // throttle count shouldn't change after the scan is completed if new requests are within the forecast coverage
+    long throttleCountBeforeRequests = mockThrottler.throttleCount.get();
+    for (long i = 0; i <= state.time.milliseconds() + TEST_TIME_INTERVAL_IN_MS; i += TEST_TIME_INTERVAL_IN_MS) {
```

Comment:
also do forward the time until end of scan results forcast time, and ensure the results are as expected